src/test/java/com/training/assignment/tests/AssignmentTest.java

```diff
@@ -139,4 +142,26 @@ public void testOopsConcept() {
         softAssert.assertThat(director.getWorkDetail())
                 .as("Work of actor and director differs").isEqualTo("Acting Directing");
     }
+
+    @Test
+    public void testAbstractInterfaceConcept() {
+        Employee employee = new Employee();
+        Yes1 yes1 = new Yes1();
+        Yes2 yes2 = new Yes2();
+
+        softAssert.assertThat(employee.getName("Shrikant"))
+                .as("Employee name doesn't match,try something else").isEqualTo("Shrikant");
+
+        softAssert.assertThat(employee.getName("Shrikant", "Sajjanshetti"))
+                .as("Employee name doesnt match,try something else").isEqualTo("Shrikant Sajjanshetti");
+
+        softAssert.assertThat(employee.getName("Shrikant", "Sajjanshetti", "Umakant"))
+                .as("Employee name doesnt match,try something else").isEqualTo("Shrikant Sajjanshetti Umakant");
```

Comment:
So when you are hard coding same value more than one time, then you should define it in a variable, in your case Shrikant is hardcoded more than one time, so define it in a varible.
Check remaining too.